bemani/backend/mga/mga.py

```diff
@@ -0,0 +1,177 @@
+# vim: set fileencoding=utf-8
+import copy
+import base64
+from typing import Any, Dict, List
+
+from bemani.backend.mga.base import MetalGearArcadeBase
+from bemani.backend.ess import EventLogHandler
+from bemani.common import Profile, VersionConstants, Time
+from bemani.data import UserID
+from bemani.protocol import Node
+
+
+class MetalGearArcade(
+    EventLogHandler,
+    MetalGearArcadeBase,
+):
+
+    name = "Metal Gear Arcade"
+    version = VersionConstants.MGA
+
+    def __update_shop_name(self, profiledata: bytes) -> None:
+        # Figure out the profile type
+        csvs = profiledata.split(b',')
+        if len(csvs) < 2:
+            # Not long enough to care about
+            return
+        datatype = csvs[1].decode('ascii')
+        if datatype != 'PLAYDATA':
+            # Not the right profile type requested
+            return
+
+        # Grab the shop name
+        try:
+            shopname = csvs[30].decode('shift-jis')
+        except Exception:
+            return
+        self.update_machine_name(shopname)
+
+    def handle_system_getmaster_request(self, request: Node) -> Node:
+        # See if we can grab the request
+        data = request.child('data')
+        if not data:
+            root = Node.void('system')
+            root.add_child(Node.s32('result', 0))
+            return root
+
+        # Figure out what type of messsage this is
+        reqtype = data.child_value('datatype')
+        reqkey = data.child_value('datakey')
+
+        # System message
+        root = Node.void('system')
+
+        if reqtype == "S_SRVMSG" and reqkey == "INFO":
+            # Generate system message
```

Comment:
Could you elaborate on what this system message should enable? I'm a bit surprised they didn't keep the same style of unlock values as Bishi Bashi, but then again they like changing things for no reason.

Reply:
Sorry, I can't answer to this. In fact, I've just dump network communications between the game and another private network, who has made basic support for MGA. My main goal, when made this plugin, was to help a friend who has the cab but was bored to being not able to save progress and has to do the (very long) tutorial each time he wanted to play the game.

Reply:
Okay, that's fine. Not sure if the values there mean anything since it doesn't match what I'm used to but somebody can fix it later.